### PR TITLE
ncm-spma: apt: Finish implementation

### PR DESCRIPTION
**Pull request by jrha**
The existing code is not sufficient to manage apt-based hosts due to major bugs and missing functionality. This PR completes the implementation and can be used to fully manage hosts.

### CODE REVIEW

File: ncm-spma/src/main/perl/spma/apt.pm [hunk 1]
```diff
@@ -77,13 +81,29 @@ Readonly my $CMD_DPKG_QUERY => [$BIN_DPKG_QUERY, qw(-W -f=${db:Status-Abbrev};${
...
+# Wrapper function for calling apt commands
+sub _call_apt
+{
+    my ($self, $cmd) = @_;
+    $self->debug(5, '_call_apt: Called with args(%s)', $cmd);
+
+    my $proc = CAF::Process->new($cmd);
+    my $output = $proc->output();
+    my $exitstatus = $? >> 8; # Get exit status from highest 8-bits
+    $self->debug(5, '_call_apt: %s exited with %s', $cmd[0] $exitstatus);
```
Comment (guillaume-philippon) [inline ncm-spma/src/main/perl/spma/apt.pm]: missing , between $cmd[0] and $exitstatus

**Comment by jrha** [comment]
It seems like this is code from a half-finished branch, apologies, please feel free to ignore for now.

**Comment by stdweird** [comment]
@jrha see also https://github.com/jrha/configuration-modules-core/pull/6

**Comment by jrha** [comment]
@hpcugentbot please test this

**Comment by stdweird** [comment]
@jrha quickly looking code looks ok. how much is this tested?

**Comment by jrha** [comment]
Right now I've been using it without issue on an Ubuntu 20.04 VM, but it's definitely early days.

**Review by stdweird — APPROVED** [review]

**Pull request merged**